### PR TITLE
letsencrypt: Add Namecheap DNS challenge support

### PR DESCRIPTION
**Pull request by nhurlock**
closes: https://github.com/home-assistant/addons/issues/1368

### CODE REVIEW

File: letsencrypt/rootfs/etc/services.d/lets-encrypt/run [hunk 1]
```diff
@@ -52,6 +52,9 @@ elif bashio::config.exists 'dns.google_creds'; then
 elif [ "${CHALLENGE}" == "dns" ] && [ "${DNS_PROVIDER}" == "dns-google-domains" ]; then
     bashio::config.require 'dns.google_domains_access_token'
     PROVIDER_ARGUMENTS+=("--authenticator" "${DNS_PROVIDER}" "--${DNS_PROVIDER}-credentials" "/data/dnsapikey" "--${DNS_PROVIDER}-propagation-seconds" "${PROPAGATION_SECONDS}")
+#Namecheap
+elif bashio::config.exists 'dns.namecheap_username' && bashio::config.exists 'dns.namecheap_api_key'; then
```
Comment (mdegat01) [inline letsencrypt/rootfs/etc/services.d/lets-encrypt/run:56]:
```suggestion
elif [ "${CHALLENGE}" == "dns" ] && [ "${DNS_PROVIDER}" == "dns-namecheap" ]; then
    bashio::config.require 'dns.namecheap_username'
    bashio::config.require 'dns.namecheap_api_key'
```
I know you probably copied what netcup did but lets borrow the model from some of the others. If the user choose namecheap lets make it fail if they missed a required value. Rather then continuing down the if statements and failing on the else and then failing when required auth info is missing.

**Review by mdegat01 — CHANGES_REQUESTED** [review]
Small change, looks good otherwise

File: letsencrypt/rootfs/etc/services.d/lets-encrypt/run (hunk 1 above)
Comment (nhurlock) [reply letsencrypt/rootfs/etc/services.d/lets-encrypt/run]: Makes sense to me, updates made 👍

**Comment by alistair23** [comment]
Gentle ping, this would be great to have

**Review by alistair23 — APPROVED** [review]

**Comment by Darsstar** [comment]
@mdegat01 This would be relly nice to have, and the change you requested seems to have been made.

**Comment by IGx89** [comment]
Was just looking for this functionality myself. Would help me if this could be merged soon!

File: letsencrypt/Dockerfile [hunk 2]
```diff
@@ -11,6 +11,7 @@ ARG \
     CERTBOT_DNS_DIRECTADMIN_VERSION \
     CERTBOT_DNS_DUCKDNS_VERSION \
     CERTBOT_DNS_GOOGLE_DOMAINS_VERSION \
+    CERTBOT_DNS_NAMECHEAP_VERSION \
     CERTBOT_DNS_HETZNER_VERSION \
     CERTBOT_DNS_INWX_VERSION \
```
Comment (agners) [inline letsencrypt/Dockerfile:16]:
```suggestion
    CERTBOT_DNS_HETZNER_VERSION \
    CERTBOT_DNS_INWX_VERSION \
    CERTBOT_DNS_NAMECHEAP_VERSION \
```

File: letsencrypt/build.yaml [hunk 3]
```diff
@@ -18,6 +18,7 @@ args:
   CERTBOT_DNS_INWX_VERSION: 2.2.0
   CERTBOT_DNS_GOOGLE_DOMAINS_VERSION: 0.1.11
   CERTBOT_DNS_TRANSIP_VERSION: 0.5.2
+  CERTBOT_DNS_NAMECHEAP_VERSION: 1.0.0
```
Comment (agners) [inline letsencrypt/build.yaml:21]:
```suggestion
  CERTBOT_DNS_NAMECHEAP_VERSION: 1.0.0
  CERTBOT_DNS_GOOGLE_DOMAINS_VERSION: 0.1.11
  CERTBOT_DNS_TRANSIP_VERSION: 0.5.2
```

File: letsencrypt/rootfs/etc/services.d/lets-encrypt/run [hunk 4]
```diff
@@ -56,6 +56,11 @@ elif [ "${CHALLENGE}" == "dns" ] && [ "${DNS_PROVIDER}" == "dns-google" ]; then
 elif [ "${CHALLENGE}" == "dns" ] && [ "${DNS_PROVIDER}" == "dns-google-domains" ]; then
     bashio::config.require 'dns.google_domains_access_token'
     PROVIDER_ARGUMENTS+=("--authenticator" "${DNS_PROVIDER}" "--${DNS_PROVIDER}-credentials" "/data/dnsapikey" "--${DNS_PROVIDER}-propagation-seconds" "${PROPAGATION_SECONDS}")
+#Namecheap
+elif [ "${CHALLENGE}" == "dns" ] && [ "${DNS_PROVIDER}" == "dns-namecheap" ]; then
+    bashio::config.require 'dns.namecheap_username'
+    bashio::config.require 'dns.namecheap_api_key'
+    PROVIDER_ARGUMENTS+=("--authenticator" "${DNS_PROVIDER}" "--${DNS_PROVIDER}-credentials" /data/dnsapikey "--${DNS_PROVIDER}-propagation-seconds" "${PROPAGATION_SECONDS}")
```
Comment (agners) [inline letsencrypt/rootfs/etc/services.d/lets-encrypt/run:63]:
```suggestion
    PROVIDER_ARGUMENTS+=("--authenticator" "${DNS_PROVIDER}" "--${DNS_PROVIDER}-credentials" "/data/dnsapikey" "--${DNS_PROVIDER}-propagation-seconds" "${PROPAGATION_SECONDS}")
```

**Review by agners — COMMENTED** [review]
Some minor nits.

**Review by agners — APPROVED** [review]

**Pull request merged**